bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/HomeNodeHandler.java

```diff
@@ -0,0 +1,134 @@
+import org.openhab.core.thing.Thing;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HomeNodeHandler} is the base class for handler of home node things.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public abstract class HomeNodeHandler extends ApiConsumerHandler {
+    private final Logger logger = LoggerFactory.getLogger(HomeNodeHandler.class);
+
+    public HomeNodeHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    void initializeProperties(Map<String, String> properties) throws FreeboxException {
+        HomeNode node = getManager(HomeManager.class).getHomeNode(getClientId());
+
+        // Gets the lowest refresh time or else, we'll keep configuration default
+        node.showEndpoints().stream().filter(ep -> ep.epType() == EpType.SIGNAL).filter(ep -> ep.refresh() != 0)
+                .min(Comparator.comparing(Endpoint::refresh)).map(Endpoint::refresh).ifPresent(rate -> {
+                    Configuration thingConfig = editConfiguration();
+                    thingConfig.put(ApiConsumerConfiguration.REFRESH_INTERVAL, Integer.toString(rate / 1000));
+                    updateConfiguration(thingConfig);
+                });
+
+        properties.putAll(node.props());
+
+        getThing().getChannels().forEach(channel -> {
+            Configuration conf = channel.getConfiguration();
+            node.type().endpoints().stream().filter(ep -> ep.name().equals(channel.getUID().getIdWithoutGroup()))
+                    .forEach(endPoint -> conf.put(endPoint.epType().asConfId(), endPoint.id()));
+            internalConfigureChannel(channel.getUID().getIdWithoutGroup(), conf, node.type().endpoints());
+        });
+    }
+
+    protected void internalConfigureChannel(String channelId, Configuration conf, List<Endpoint> endpoints) {
+    }
+
+    @Override
+    protected void internalPoll() throws FreeboxException {
+        HomeManager homeManager = getManager(HomeManager.class);
+        getThing().getChannels().stream().filter(channel -> isLinked(channel.getUID())).forEach(channel -> {
+            State result = UnDefType.UNDEF;
+            Integer slotId = getSlotId(channel.getConfiguration(), EpType.SIGNAL.asConfId());
+            if (slotId instanceof Integer) {
+                try {
+                    EndpointState state = homeManager.getEndpointsState(getClientId(), slotId);
+                    if (state != null) {
+                        result = getChannelState(homeManager, channel.getUID().getIdWithoutGroup(), state);
+                    } else {
+                        result = getChannelState(homeManager, channel.getUID().getIdWithoutGroup());
+                    }
+                } catch (FreeboxException e) {
+                    logger.info("Error updating channel : {}", e.getMessage());
```

Comment:
This should be debug if it is self-healing or warn if it is fatal.

Reply:
Changed.